pkg/autodiscovery/listeners/cloudfoundry.go

```diff
@@ -79,11 +80,18 @@ func (l *CloudFoundryListener) Listen(newSvc chan<- Service, delSvc chan<- Servi
 	}()
 }
 
+func (l *CloudFoundryListener) GetRefreshCount() int64 {
+	l.RLock()
+	defer l.RUnlock()
+	return l.refreshCount
+}
+
```

Comment:
We should use the atomic package instead of lock for this I think.

Also why are we creating a public method for this ? Could we not just access it from the tests ?

Reply:
It's true that we actually only need to access it from the tests, so I'll just remove the getter and retrieve the value in a proper way in the test suite.